webClient/src/app/core/menu-bar/menu-bar.component.ts

```diff
@@ -567,19 +567,51 @@ export class MenuBarComponent implements OnInit, OnDestroy {
     this.editorControl.toggleTree.next();
   }
 
-  closeAll() {
-    let closeAllRef;
-    if (this.fileCount == 0) { //TODO: Enhance such that closeAll not visible if no tabs are open
+  promptToSave(file: ProjectContext): Promise<any>{
+    return new Promise((resolve, reject) => {
+      if(file.changed) {
+        const title = 'Do you want to save the changes you made to \'' + file.name + '\?';
```

Comment:
In most instances of Editor, we've been formatting file names as 'file-name.txt'

Reply:
I can save a fine in editor with any extenstion. Let FileName.abc
then it would be inappropriate to mention FileName.abc.txt in the prompt